src/sentry/static/sentry/app/utils/cursorPoller.tsx

```diff
@@ -0,0 +1,104 @@
+import {Client, Request} from 'app/api';
+import parseLinkHeader from 'app/utils/parseLinkHeader';
+
+type Options = {
+  endpoint: string;
+  success: (data: any, link?: string | null) => void;
+};
+
+const BASE_DELAY = 3000;
+const MAX_DELAY = 60000;
+
+class CursorPoller {
+  constructor(options: Options) {
+    this.options = options;
+    this.pollingEndpoint = options.endpoint;
+  }
+
+  api = new Client();
+  options: Options;
+  pollingEndpoint: string;
+  timeoutId: number | null = null;
+  lastRequest: Request | null = null;
+  active: boolean = true;
+
+  reqsWithoutData = 0;
```

Comment:
Should these all be marked as `private` to get compiler errors on external usage?